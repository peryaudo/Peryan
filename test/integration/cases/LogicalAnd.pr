func lhs(ty, b) :: Bool {
	mes "lhs for " + ty
	return b
}

func rhs(ty::String, b::Bool) :: Bool {
	mes "rhs for " + ty
	return b
}

if lhs("and", true) & rhs("and", true) {
	mes "total true"
}


if lhs("and", false) & rhs("and", true) {
	mes "total false"
}

if lhs("or", true) | rhs("or", false) {
	mes "total true"
}

if lhs("or", false) | rhs("or", false) {
	mes "total false"
}
