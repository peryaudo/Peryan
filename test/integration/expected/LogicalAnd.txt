lhs for and
rhs for and
total true
lhs for and
lhs for or
total true
lhs for or
rhs for or
